package.json

```diff
@@ -7,7 +7,7 @@
     "html-elements": "tools/html-elements.js"
   },
   "scripts": {
-    "build": "pnpm build --prefix src/web --",
+    "build": "pnpm turbo run build",
```

Comment:
Can you explain this change, I don't understand.  How does it know the build directory to use when running this?

Reply:
It basically goes inside all the other `package.json` specified by pnpm workspace to find a `build` command.

> Each key in the pipeline object is the name of a task that can be executed by turbo run. If turbo finds a workspace package with a package.json scripts object with a matching key, it will apply the pipeline task configuration to that NPM script during execution.

https://turborepo.org/docs/reference/configuration#pipeline

If for example we had scripts for `lint` and `test` in our microservices' `package.json`, running `pnpm turbo run lint test build` would run all those tasks in parallel when possible.

Reply:
Ok, that's really helpful.  So this means that we could change the way we do things a bit.  Previously, we always thought about running stuff like lint, test, etc. from the top-level of the repo across all of the projects in the tree.  With this, we could be more specific to each sub-project, and then use turborepo to figure out what to run.

I also wonder if we could use this to `build` our Docker images?  That would be cool.  Is turborepo able to figure out when a sub-project's files have been update, or is this only happening at the level of the command being run?

Reply:
I am also wondering about how to handle the Docker images with Turborepo. I haven't found much info on it and I don't have enough Docker experience yet.

> Is turborepo able to figure out when a sub-project's files have been update, or is this only happening at the level of the command being run?

Yes, it is able to see when sub-project's files have been updated.